packages/access-client/src/access.js

```diff
@@ -0,0 +1,298 @@
+    with: space,
+    nb: {
+      delegations: Object.fromEntries(entries),
+    },
+    // must be embedded here because it's referenced by cid in .nb.delegations
+    proofs: [...delegations, ...proofs],
+  })
+
+  return out
+}
+
+/**
+ * Requests specified `access` level from specified `account`. It will invoke
+ * `access/authorize` capability and keep polling `access/claim` capability
+ * until access is granted or request is aborted.
+ *
+ * @param {API.Agent} agent
+ * @param {object} input
+ * @param {API.AccountDID} input.account
+ * @param {API.ProviderDID} [input.provider]
+ * @param {API.DID} [input.audience]
+ * @param {API.Access} [input.access]
+ * @returns {Promise<API.Result<PendingAccessRequest, API.AccessAuthorizeFailure|API.InvocationError>>}
+ */
+export const request = async (
+  agent,
+  {
+    account,
+    provider = /** @type {API.ProviderDID} */ (agent.connection.id.did()),
+    audience = agent.did(),
+    access = spaceAccess,
+  }
+) => {
+  // Request access from the account.
+  const { out: result } = await agent.invokeAndExecute(Access.authorize, {
+    audience: DID.parse(provider),
+    with: audience,
+    nb: {
+      iss: account,
+      // New ucan spec moved to recap style layout for capabilities and new
+      // `access/request` will use similar format as opposed to legacy one,
+      // in the meantime we translate new format to legacy format here.
+      att: [...toCapabilities(access)],
+    },
+  })
+
+  return result.error
+    ? result
+    : {
+        ok: new PendingAccessRequest({
+          ...result.ok,
+          agent,
+          audience,
+          provider,
+        }),
+      }
+}
+
+/**
+ * Claims access that has been delegated to the given audience, which by
+ * default is the agent's DID.
+ *
+ * @param {API.Agent} agent
+ * @param {object} input
+ * @param {API.DID} [input.audience]
+ * @param {API.ProviderDID} [input.provider]
+ * @returns {Promise<API.Result<GrantedAccess, API.AccessClaimFailure|API.InvocationError>>}
+ */
+export const claim = async (
+  agent,
+  {
+    provider = /** @type {API.ProviderDID} */ (agent.connection.id.did()),
+    audience = agent.did(),
+  } = {}
+) => {
+  const { out: result } = await agent.invokeAndExecute(Access.claim, {
+    audience: DID.parse(provider),
+    with: audience,
+  })
+
+  if (result.error) {
+    return result
+  } else {
+    const delegations = Object.values(result.ok.delegations)
+    const proofs = delegations.flatMap((proof) => bytesToDelegations(proof))
+    return { ok: new GrantedAccess({ agent, provider, audience, proofs }) }
+  }
+}
+
+/**
+ * Represents a pending access request. It can be used to poll for the requested
+ * delegation.
+ */
+class PendingAccessRequest {
+  /**
+   * @param {object} source
+   * @param {API.Agent} source.agent
+   * @param {API.DID} source.audience
+   * @param {API.ProviderDID} source.provider
+   * @param {number} source.expiration
```

Comment:
what kind of `number`? e.g. is it number of second or milliseconds since epoch? something else?
(should it be a `Date`?)